src/pnpq/apt/connection.py

```diff
@@ -155,44 +176,52 @@ def __post_init__(self) -> None:
 
         self.log.debug("Finishing connection post-init...")
 
-    # TODO from a multi-threading point of view, it might be much
-    # easier to assume that, for the lifetime of a program, a
-    # connection is held by a single object that cannot be
-    # closed. However, we should still probably implement a context
-    # manager for this object.
-    #
-    # The tricky part is managing the threads; they will all need to
-    # be paused before closing the serial connection... and then how
-    # do we clean up the child threads if this object gets cleaned up?
+    def close(self) -> None:
 
-    # def __enter__(self) -> None:
-    #     self.open()
+        self.send_message_no_reply(
+            AptMessage_MGMSG_HW_STOP_UPDATEMSGS(
+                destination=Address.GENERIC_USB,
+                source=Address.HOST_CONTROLLER,
+            )
+        )
 
-    # def __exit__(self, exc_type, exc_value, exc_tb) -> None:
-    #     self.close()
+        time.sleep(1)
 
-    # def open(self) -> None:
-    #     self.connection.open()
+        self.stop_event.set()
 
-    # def close(self) -> None:
-    #     self.connection.flush()
-    #     self.connection.close()
+        self.tx_ordered_sender_queue.shutdown()
+        self.tx_ordered_sender_thread.join()
+
+        self.connection.flush()
+        self.connection.close()
+
+        self.rx_dispatcher_thread.join()
+
+        self.log.debug("Successfully closed the APTConnection.")
 
     def rx_dispatch(self) -> None:
         with self.rx_dispatcher_thread_lock:
-            while True:
+            while not self.stop_event.is_set():
+
                 partial_message: None | AptMessageForStreamParsing = None
                 full_message: Optional[AptMessage] = None
                 try:
                     message_bytes = self.connection.read(6)
+                # Serial bus not connected error
+                except Exception:  # pylint: disable=W0718
```

Comment:
Probably a debug-level log of this exception is a good idea. You can use `exc_info` to log the exception.